### DISCUSSION TITLE
Bug/Issue: You are running the legacy version of PnP PowerShell. This version will be archived soon which means that while staying available, no updates or fixes will be released. Consider installing the newer prereleased cross-platform version of PnP PowerShell. This version has numerous improvements and many more cmdlets available. To install the new version: Uninstall-Module -Name SharePointPnPPowerShellOnline -AllVersions -Force Install-Module -Name PnP.PowerShell Read more about the new cross-platform version of PnP PowerShell at https://pnp.github.io/powershell To hide this message set the environment variable PNPLEGACYMESSAGE to "false" In PowerShell add $env:PNPLEGACYMESSAGE='false' to your profile. Alternatively use 'Connect-PnPOnline -Url [yoururl] -WarningAction Ignore'

### DISCUSSION THREAD

**Discussion by Anto4595** [pnp/powershell#4450] [General]
Hello @erwinvanhunen 

I tried to install the Latest 2.99.16 nighty package in azure automation and I am receiving this warning message .

You are running the legacy version of PnP PowerShell. This version will be archived soon which means that while staying available, no updates or fixes will be released. Consider installing the newer prereleased cross-platform version of PnP PowerShell. This version has numerous improvements and many more cmdlets available. To install the new version: Uninstall-Module -Name SharePointPnPPowerShellOnline -AllVersions -Force Install-Module -Name PnP.PowerShell Read more about the new cross-platform version of PnP PowerShell at https://pnp.github.io/powershell To hide this message set the environment variable PNPLEGACYMESSAGE to "false" In PowerShell add $env:PNPLEGACYMESSAGE='false' to your profile. Alternatively use 'Connect-PnPOnline -Url [yoururl] -WarningAction Ignore' 

Can you Please advise on this ?

or Can you release the Latest version in the Browse gallery available with 7.1 runtime version.

Thank You.

**Comment by gautamdsheth** [comment]
Hi, we will not be doing that. 
Version 2.x which is the latest major release works on PowerShell 7.2 or later.
The upcoming version 3.x will be supported on PowerShell 7.4 or later.

Azure automation supports PS 7.2 , so would advise you to use 2.12.0 major version .

**Comment by jackpoz** [comment]
What command did you use to install it ? Because that error message is found only in https://github.com/pnp/PnP-PowerShell which is an old version which corresponds to https://www.powershellgallery.com/packages/SharePointPnPPowerShellOnline , while the new version is https://www.powershellgallery.com/packages/PnP.PowerShell

**Reply by gautamdsheth** [reply]
Yes, as @jackpoz said that is an old deprecated version, you should use the latest 2.12 version in Azure automation.

You can find installation instructions here:

https://pnp.github.io/powershell/articles/installation.html

**Comment by Anto4595** [comment]
Hello @gautamdsheth @jackpoz In azure automation account I have installed the version 2.12.0 which is available from browse gallery and one of the script that is running in that automation account gives me a warning that there is a newer version 2.99.16 available and to install it.

But this version is not available in Browse gallery. So I tried to install it using deploy in Azure automation from the pnp.powershell link and it was installed and then I run the script and it was showing the message that I have added in the title.

Then I deleted the 2.99.16 version and checked and it shows again the warning to install 2.99.16 this is the behaviour.


I am also getting SSL Connection error. Check inner exception Do we know why we getting this?

In the azure automation browse gallery I have installed the pnp.powershell and sharepointpnp.powershell versions

**Reply by jackpoz** [reply]
You can ignore the warning related to 2.99.16, as that's a nightly build version.
You don't need SharePointPnPPowerShellOnline and installing that might cause issues, so I would suggest to uninstall it and have only PnP.PowerShell installed.

Maybe the SSL error is caused by having both versions, so try after cleaning things up.

**Reply by Anto4595** [reply]
@jackpoz The script that I am running in Azure Automation account gets the list of the virtual networks and create a excel file and store it in a sharepoint folder.

So at that time I installed the pnp.powershell and sharepointpnp as well. I have uninstalled the sharepointpnp now and I will notice it for couple of days and let you know if I am getting the ssl issue.

**Comment by Anto4595** [comment]
Hello @jackpoz @gautamdsheth 

I removed the sharepointpnppowershellonline modules and had only the pnp.powershell 2.12.0 version

Today I got the same error.

![image](https://github.com/user-attachments/assets/8f1174fd-26d9-4fe7-b843-b15d1b485c8f)

I am getting in the line 54 But in my script the Line 54 has 

$null = Set-AzContext -Subscription $subscription.SubscriptionID

As I said before this error occurs appears occasionally.

**Reply by jackpoz** [reply]
That seems related to Set-AzContext , a different PowerShell module, so not much related to PnP PowerShell
